lib/rucio/core/replica.py

```diff
@@ -2261,3 +2263,130 @@ def mark_unlocked_replicas(rse, bytes, session=None):
             total_bytes += bytes
             rows.append({'scope': scope, 'name': name})
     return rows
+
+
+@transactional_session
+def get_cleaned_updated_collection_replicas(total_workers, worker_number, session=None):
+    """
+    Get update request for collection replicas.
+    :param total_workers:      Number of total workers.
+    :param worker_number:      id of the executing worker.
+    :param session:            Database session in use.
+    :returns:                  List of ids of update requests for collection replicas.
+    """
+    # Delete update requests which do not have collection_replicas and delete duplicates
+    replica_update_requests = session.query(models.UpdatedCollectionReplica).all()
+    update_requests_with_rse_id = []
+    update_requests_without_rse_id = []
+    for update_request in replica_update_requests:
+        if update_request.rse_id is not None:
+            small_request = {'name': update_request.name, 'scope': update_request.scope, 'rse_id': update_request.rse_id}
+            if small_request not in update_requests_with_rse_id:
+                update_requests_with_rse_id.append(small_request)
+            else:
+                update_request.delete(session=session)
+                continue
+            found_replicas = session.query(models.CollectionReplica).filter(models.CollectionReplica.rse_id == update_request.rse_id, models.CollectionReplica.scope == update_request.scope, models.CollectionReplica.name == update_request.name).all()
```

Comment:
you can cut line length, just put a backslash and start the next line with the dot of the function

(happens a couple of times, you can do it everywhere)